LBH.AdultSocialCare.Api/V1/Gateways/SupplierGateways/SupplierGateway.cs

```diff
@@ -37,11 +36,17 @@ public async Task<SupplierDomain> CreateAsync(Supplier supplier)
             }
         }
 
-        public async Task<IEnumerable<SupplierDomain>> ListAsync()
+        public async Task<PagedList<SupplierDomain>> ListAsync(RequestParameters parameters)
         {
-            var res = await _databaseContext.Suppliers
+            var suppliersCount = await _databaseContext.Suppliers
+                .CountAsync().ConfigureAwait(false);
+            var suppliers = await _databaseContext.Suppliers
                 .ToListAsync().ConfigureAwait(false);
-            return res?.ToDomain();
+
+            var suppliersPage = suppliers.GetPage(parameters.PageNumber, parameters.PageSize);
```

Comment:
Can this extend the suppliers IQueryable to only get the records in the page range straight from DB? I think would be a slightly better approach than loading all suppliers in memory before only taking a slice of the response.